core/src/main/java/io/kestra/core/services/ExecutionService.java

```diff
@@ -506,4 +511,28 @@ private Set<String> taskRunWithAncestors(Execution execution, List<TaskRun> task
             .flatMap(throwFunction(taskRun -> this.getAncestors(execution, taskRun).stream()))
             .collect(Collectors.toSet());
     }
+
+    /**
+     * This method is used to retrieve previous existing execution
+     * @param flow The flow of the failed execution
+     * @param execution The failed execution
+     * @return The next retry date, null if maxAttempt || maxDuration is reached
+     */
+    public Instant nextRetryDate(Flow flow, Execution execution) {
+        Execution originalExecution = executionRepository.findById(execution.getTenantId(), execution.getParentId()).orElse(null);
```

Comment:
It is not a good idea to use the ExecutionRepository inside the executor, it could lead to issues with indexing delay with the Kafka implementation.